activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -303,7 +303,7 @@ def null_scope?
 
       def find_from_target?
         loaded? ||
-          owner.strict_loading? ||
+          (owner.strict_loading? && owner.strict_loading_all?) ||
```

Comment:
If `strict_loading_all?` returns `true`, `strict_loading?` is always `true`?
```suggestion
          owner.strict_loading_all? ||
```

Reply:
Ah never mind. You can call `strict_loading!(false, mode: :all)` and the `mode` will still be set to `:all`.